src/MergeContext.js

```diff
@@ -954,12 +968,47 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // recalculates this._stagingFailed value for PRs without staging commit
+    async _recalculateStagingFailed() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // we consider all commits created by the bot user and referencing this PR as staged commits
+        let stagedEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagedEvents.length)
+            return;
+
+        // just in case: events should be already sorted by date
+        stagedEvents = stagedEvents.sort((ev1, ev2) => Date.parse(ev1.created_at) - Date.parse(ev2.created_at));
+        const lastStaged = stagedEvents[stagedEvents.length - 1];
+        const mergeCommit = await this._getMergeCommit();
+        const mergeCommitCreatedAt = Date.parse(mergeCommit.author.date);
+        const stagedCommitCreatedAt = Date.parse(lastStaged.created_at);
+        assert(mergeCommitCreatedAt);
+        assert(stagedCommitCreatedAt);
+        // whether the merge commit is fresher than the last staged commit
+        if (stagedCommitCreatedAt < mergeCommitCreatedAt)
+            return false;
```

Comment:
The "false" return value seems to be unnecessary/undefined/unused. Should not we set the filed we were asked to recalculate instead of returning false? Let's also clarify why we do not care about the equality case. Adjust as needed, of course.

```suggestion
        // the staging event for merge commit X is always created after X so we do not
        // need to deal with the lastStagingEventCreatedAt == mergeCommitCreatedAt case
        assert(lastStagingEventCreatedAt != mergeCommitCreatedAt);
        if (lastStagingEventCreatedAt < mergeCommitCreatedAt) {
            // merge commit created after the last staging event
            this._stagingFailed = true;
            return; 
        }
```